lib/migration_tools/tasks.rb

```diff
@@ -29,7 +29,12 @@ def migrations_paths
 
     def migrator(target_version = nil)
       if ActiveRecord::VERSION::MAJOR > 3
-        ActiveRecord::Migrator.new(:up, ActiveRecord::Migrator.migrations(migrations_paths), target_version)
+        migrations = if defined?(::ActiveRecord::MigrationContext)
+          ActiveRecord::MigrationContext.new(migrations_paths).migrations
+        else
+          ActiveRecord::Migrator.migrations(migrations_paths)
+        end
+        ActiveRecord::Migrator.new(:up, migrations, target_version)
       else
         ActiveRecord::Migrator.new(:up, migrations_paths, target_version)
       end
```

Comment:
We could make an almost verbatim copy of https://github.com/zendesk/active_record_shards/blob/69c291bc2400d51489cd07a18578f43b097d9b73/test/helper.rb#L92-L103 – I like that it has only one level of conditionals.